File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = [0] * len(prices)
    stack = []
    
    for i, p in enumerate(prices):
        while  stack != [] and p < prices[stack[-1]]:
            j = stack.pop()
            answer[j] = i - j
        stack.append(i)
        
    while stack != []:
        k = stack.pop()
        answer[k] = len(prices) - 1 - k
        
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer(prices.size(), 0);
    stack<int> stk;
    
    for (int i=0; i<prices.size(); i++){
        while (stk.size() != 0 and prices[i] < prices[stk.top()]){
            answer[stk.top()] = i - stk.top();
            stk.pop();
        }
        stk.push(i);
    }
    while (!stk.empty()){
        answer[stk.top()] = prices.size() - stk.top() - 1;
        stk.pop();
    }
    return answer;
}